pkgs/tools/misc/hdfview/default.nix

```diff
@@ -2,29 +2,34 @@
 
 stdenv.mkDerivation rec {
   pname = "hdfview";
-  version = "3.1.3";
+  version = "3.1.4";
 
   src = fetchurl {
     url = "https://support.hdfgroup.org/ftp/HDF5/releases/HDF-JAVA/${pname}-${version}/src/${pname}-${version}.tar.gz";
-    sha256 = "sha256-VmgHSVMFoy09plU5pSnyaPz8N15toy7QfCtXI7mqDGY=";
+    sha256 = "sha256-iY/NUifU57aX795eWpaUBflrclF/nfvb3OKZEpD9VqA=";
   };
 
   nativeBuildInputs = [
-    ant jdk
-    nettools  # "hostname" required
+    ant
+    jdk
+    nettools # "hostname" required
```

Comment:
(FWIW there's actually a special `hostname` package which should be used in most cases instead, because the correct underlying package varies by os)

_That said_, with either of these when I try building this on nixos aarch64 (emulated) I get:

```
/build/hdfview-3.1.4/build.xml:146: Execute failed: java.io.IOException: Cannot run program "hostname" (in directory "/build/hdfview-3.1.4"): error=0, Failed to exec spawn helper: pid: 108, exit value: 1
```

If you dig in to `build.xml` it looks like all that's being done with `hostname` is attempting to detect building on an ubuntu machine.

```xml
<exec executable="hostname" outputproperty="computer.hostname"/>
<condition property="isUbuntu">
<contains string="${computer.hostname}" substring="ubuntu"/>
</condition>
```

I would suggest replacing this block entirely with e.g.

```xml
<condition property="isUbuntu" value="false"/>
```

through patching. `hostname` is never going to give a real answer anyway inside the sandbox as it would introduce an impurity.

This way you could drop the `hostname` requirement entirely.

Reply:
A careful observation and an elegant solution!